gnd/src/main/java/com/google/android/gnd/persistence/local/LocalDataStore.java

```diff
@@ -164,6 +164,18 @@
   /** Returns all queued, failed, and completed offline areas from the local data store. */
   Flowable<ImmutableList<OfflineBaseMap>> getOfflineAreasOnceAndStream();
 
+  /** Delete an offline area and its *unique* tiles. */
+  Completable deleteOfflineArea(String offlineAreaId);
+
   /** Returns the offline area with the specified id. */
   Single<OfflineBaseMap> getOfflineAreaById(String id);
+
+  /**
+   * Update the area count of an existing tile source in the local data store with the area count of
+   * {@param tilesource}.
+   */
+  Single<TileSource> updateTileSourceAreaCountByUrl(TileSource tilesource);
```

Comment:
I'd expect this to accept the URL and new count; this makes what's happening more obvious and reduces the surface area of the internal API.

Also, returning a `Completable` here may be more correct.

Reply:
done, thanks!